src/backend/api/redis.py

```diff
@@ -48,20 +52,66 @@ def set_mirrors_to_cache(
     key = str(key)
     redis_engine = RedisEngine.get_instance()
     mirrors = json.dumps(mirrors, cls=DataClassesJSONEncoder)
-    redis_engine.set(
+    await redis_engine.set(
         key,
         mirrors,
         CACHE_EXPIRED_TIME,
     )
+    await redis_engine.close()
+    await redis_engine.connection_pool.disconnect()
 
 
-def get_url_types_from_cache() -> List[AnyStr]:
+async def get_geolocation_from_cache(key: AnyStr) -> Optional[Dict]:
+    key = str(key)
+    redis_engine = RedisEngine.get_instance()
+    coords = await redis_engine.get(key)
+    await redis_engine.close()
+    await redis_engine.connection_pool.disconnect()
+    if coords:
+        return json.loads(coords)
+
+
+async def set_geolocation_to_cache(key: AnyStr, coords: Tuple) -> None:
```

Comment:
FYI: python3.9 supports built-in types for typing)

Reply:
Thanks :)